src/toolkitAPI.c

```diff
@@ -1248,6 +1248,59 @@ int DLLEXPORT swmm_getSystemRunoffStats(SM_RunoffTotals *runoffTot)
     return(errorcode);
 }
 
+int DLLEXPORT swmm_getGagePrecip(int index, double *value)
```

Comment:
I think you should output some more interesting values here. Take a look at this function inside `gage.c`

`double gage_getPrecip(int j, double *rainfall, double *snowfall)`

This function provides three results.  If you call it directly from here, you can fetch three values:

1. rainfall
2. snow
3. total precip

I suggest your function do this:

```C
int DLLEXPORT swmm_getGagePrecip(int index, double *rainfall, double *snowfall, double *total)
```

Users can choose to ignore the values they do not need. If you do it this way, we have a lot more options!

Reply:
By snow, do you mean the snow reading from the subcatchment associated with the gate
```
double        newSnowDepth;    // current snow depth (ft)
```